### PR TITLE
build: fix image name when having the case foo:1234/bar

### PR DESCRIPTION
**Pull request by cpanato**
## Description
Fixes the issue described in https://github.com/openfaas/faas-cli/issues/875

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
Fixes: https://github.com/openfaas/faas-cli/issues/875

## How Has This Been Tested?

Build this PR, and you can follow the steps described in the issue https://github.com/openfaas/faas-cli/issues/875

```
$ faas-cli build --tag sha --image foo:1234/bar --handler ./foo --name foo --lang go
....
#29 writing image sha256:3010593902cbc6c65947e710e240e4f40f25533b5721628a94a25a4a2313245b done
#29 naming to foo:1234/bar:latest-a26c44e done
#29 DONE 0.0s
Image: foo:1234/bar:latest-a26c44e built.
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

cc @alexellis

### CODE REVIEW

File: schema/image_test.go [hunk 1]
```diff
@@ -9,6 +9,13 @@ func Test_BuildImageName_DefaultFormat(t *testing.T) {
 	if got != want {
 		t.Errorf("BuildImageName want: \"%s\", got: \"%s\"", want, got)
 	}
+
+	want = "foo:123d/img:latest"
```
Comment (alexellis) [inline schema/image_test.go:13]: I would like to see new tests for new scenarios.

by the way, `foo:123d/img:latest` looks like an invalid image tag. 🤔

Comment (cpanato) [reply schema/image_test.go:13]: in this case that produce the correct, because for that test it is using the `default` and not the `--tag sha` option

Comment (alexellis) [reply schema/image_test.go:13]: `foo:123d` looks like an invalid prefix to me

Comment (cpanato) [reply schema/image_test.go:13]: i was trying to reproduce the item described in the issue


```
$ faas-cli build --tag=sha --lang python3 --image foo:4567/bar --handler . --name my-function
[...]
Successfully tagged foo:4567/bar-52f0c55:latest
Image: foo:4567/bar-52f0c55 built.
```

the `foo:123d` can be whatever I guess

**Review by LucasRoesler — APPROVED** [review]
Looks good :shipit:

File: schema/image.go [hunk 2]
```diff
@@ -65,7 +65,8 @@ func (i *BuildFormat) Set(value string) error {
 // BuildImageName builds a Docker image tag for build, push or deploy
 func BuildImageName(format BuildFormat, image string, version string, branch string) string {
 	imageVal := image
-	if strings.Contains(image, ":") == false {
+	splitImage := strings.Split(image, "/")
```
Comment (alexellis) [inline schema/image.go:68]: I think I'd have used `strings.Index()` here in both cases, and used that for the operations.

If the tests pass, this is fine to keep as is.

**Review by alexellis — APPROVED** [review]
Approved

**Comment by alexellis** [comment]
Can you check if the `--extra-tag` flag is also covered for `faas-cli publish --extra-tag latest`?

**Comment by cpanato** [comment]
@alexellis 

```
$ faas-cli publish --extra-tag latest --tag sha --yaml foo.yml
...
Publishing with command: docker [buildx build --progress=plain --platform=linux/amd64 --output=type=registry,push=true --tag registry:3000/test/foo:latest-aa4f2f4 . --tag registry:3000/test/foo:latest]
```

**Pull request merged**